### PR TITLE
CU-2t2ew2w - vAMM integration tests

### PR DESCRIPTION
**Pull request by 0xangelo**
## Issue
[CU-2t2ew2w](https://app.clickup.com/t/2t2ew2w)

## Description
This implements additional tests which use all real pallets, including the Oracle (Apollo) and vAMM. It does not yet cover all functionality of the Clearing House that involves the other two pallets

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2t2ew2w Integration tests with Vamm Pallet](https://app.clickup.com/t/2t2ew2w)

File: frame/composable-traits/src/vamm.rs [hunk 1]
```diff
@@ -78,7 +78,7 @@ pub trait Vamm {
 }
 
 /// Specify a common encapsulation layer for the [`create`](Vamm::create) function.
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Debug, Decode, Default, Encode, MaxEncodedLen, PartialEq, TypeInfo)]
```
Comment (0xangelo) [inline frame/composable-traits/src/vamm.rs:81]: @Cardosaum I noticed that these were necessary for building the runtime. How did you get around it in your tests?

Comment (Cardosaum) [reply frame/composable-traits/src/vamm.rs:81]: :thinking: I think it might be because I specify these constraints in the types itself on `lib.rs`?

**Comment by 0xangelo** [comment]
bors try

**Review by Cardosaum — APPROVED** [review]
LGTM

I liked how well organized are all the sections in the `mock.rs` file, well done!

File: frame/clearing-house/src/integrations/tests.rs [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+// -------------------------------------------------------------------------------------------------
+//                                  Helper Functions and Traits
+// -------------------------------------------------------------------------------------------------
+
+fn advance_blocks_by(n: BlockNumber) {
+	for _ in 0..n {
+		let curr_block = System::block_number();
+		if curr_block > 0 {
+			Timestamp::on_finalize(curr_block);
+			System::on_finalize(curr_block);
```
Comment (kevin-composable) [inline frame/clearing-house/src/integrations/tests.rs:41]: I would call `System::on_finalize` last as it is always called last. This probably won't affect functionality now but can down the line

Comment (kevin-composable) [reply frame/clearing-house/src/integrations/tests.rs:41]: same with `on_initialize` but before the other two

**Review by kevin-composable — APPROVED** [review]
One NIT comment so I am going to approve - Nice work Angelo

File: frame/clearing-house/src/integrations/tests.rs (hunk 2 above)
Comment (0xangelo) [reply frame/clearing-house/src/integrations/tests.rs:41]: Ah, thanks, totally missed that

**Pull request merged**